parser/Parser.cc

```diff
@@ -90,28 +140,33 @@ unique_ptr<Node> Parser::run(sorbet::core::GlobalState &gs, core::FileRef file,
     buffer += source;
     buffer += "\0\0"sv;
     StableStringStorage<> scratch;
-    unique_ptr<ruby_parser::base_driver> driver;
-    if (settings.traceParser) {
-        driver =
-            make_unique<ruby_parser::typedruby_debug27>(buffer, scratch, Builder::interface, !!settings.traceLexer);
-    } else {
-        driver =
-            make_unique<ruby_parser::typedruby_release27>(buffer, scratch, Builder::interface, !!settings.traceLexer);
-    }
 
-    for (string local : initialLocals) {
-        driver->lex.declare(local);
+    auto driver = makeDriver(settings, buffer, scratch, initialLocals);
+    auto ast = builder.build(driver.get(), settings.traceParser);
+
+    // Always report the original parse errors
+    errorToError(gs, file, driver->diagnostics);
+
+    if (ast != nullptr) {
```

Comment:
Since this path uses `errorToError`, but the indentation-aware path uses `reportDiagnostics`, perhaps we should just inline `errorToError`'s code here to make the symmetry between the cases more explicit?  I think that would also have the benefit of making the `onlyHints` logic more explicit, so that the reader doesn't get concerns about errors being emitted twice, only to discover that `onlyHints` would prevent that.  (Still might be good to expand on the rationale for always reporting errors?)